18F/identity-idp
# Deploy script updates for local execution

## 🛠 Summary of changes

This PR includes some updates to the deployment-related scripts in #9859 to allow them to run locally while I try and figure out when / if we can actually enable Github Actions for this repository. The changes here are mostly focused on things that got in my way while I was deployer this week as well as some general quality of life improvements, things like:

- Only checking for CI dependencies (`$GH_TOKEN`) when we're actually running in CI
- Always specify what repository we want `gh` to affect (via `--repo` arg)
- Allow adding a custom hook to format changelogs

## Testing

1. Fork identity-idp.

2. Create a `stages/prod` branch in your fork

3. Add your fork as a remote on your local clone:

```shell
git remote add upstream git@github.com:YOUR_NAME/identity-idp.git
```
4. Run `create-deploy-pr` in dry run mode:

```shell
DRY_RUN=1 GIT_REMOTE=upstream GH_REPO=YOUR_NAME/identity-idp scripts/create-deploy-pr 
```
5. Verify everything looks good and re-run with DRY_RUN disabled:

```shell
GIT_REMOTE=upstream GH_REPO=YOUR_NAME/identity-idp scripts/create-deploy-pr 
```

6. Note the number assigned by Github to your PR, then merge it

7. Run `create-release` in dry run mode:

```shell
DRY_RUN=1 GH_REPO=YOUR_NAME/identity-idp scripts/create-release <PR_NUMBER>
```

8. Run `create-release` again to actually create your release:

```shell
GH_REPO=YOUR_NAME/identity-idp scripts/create-release <PR_NUMBER>
```
